src/crimson/os/seastore/cache.cc

```diff
@@ -31,20 +31,27 @@ Cache::retire_extent_ret Cache::retire_extent_addr(
   Transaction &t, paddr_t addr, extent_len_t length)
 {
   LOG_PREFIX(Cache::retire_extent);
-  if (auto ext = t.write_set.find_offset(addr); ext != t.write_set.end()) {
-    DEBUGT("found {} in t.write_set", t, addr);
+  CachedExtentRef ext;
+  auto result = t.get_extent(addr, &ext);
+  if (result == Transaction::get_extent_ret::PRESENT) {
+    DEBUGT("found {} in t", t, addr);
     t.add_to_retired_set(CachedExtentRef(&*ext));
     return retire_extent_iertr::now();
-  } else if (auto iter = extents.find_offset(addr);
-      iter != extents.end()) {
-    auto ret = CachedExtentRef(&*iter);
+  } else if (result == Transaction::get_extent_ret::RETIRED) {
+    ERRORT("{} is already retired", t, addr);
+    ceph_abort();
```

Comment:
This check seems to be the only change in behavior beyond a refactor to use Transaction::get_extent.  Is that correct?

Reply:
My original intention is to leverage `Transaction::get_extent()` also lookup `Transaction::read_set`, if present, add them to `Transaction::retired_set`.

After a second thought, I agree with you that the only behavior change of this commit is the RETIRED check here. So now I think it is mostly a cleanup and a pre-requisite commit to add the missing `t.add_to_read_set(ext)' at line 48 below, see https://github.com/ceph/ceph/pull/42080/commits/dbc39791bc6b68948301b92fffe7972d53544d0a#diff-2d1edaa7a9eebb3a65135f5af648dd069934fa6bf6e6e0339b5771b0daef3727R48.

Reply:
Oh, I'm ok with the cleanup -- it's quite a bit clearer.  I just wanted to make sure I understood the patch.